js-api-doc/value/mixin.d.ts

```diff
@@ -0,0 +1,19 @@
+import {Value} from './index';
+
+/**
+ * Sass's [mixin type](https://sass-lang.com/documentation/values/mixins).
+ *
+ * @category Custom Function
+ */
+export class SassMixin extends Value {
+  /**
+   * Creates a new first-class mixin that can be invoked using
+   * [`meta.apply()`](https://sass-lang.com/documentation/modules/meta#apply).
+   *
+   * @param signature - The function signature, like you'd write for the
+   * [`@function rule`](https://sass-lang.com/documentation/at-rules/function).
+   * @param callback - The callback that's invoked when this function is called,
+   * just like for a {@link CustomFunction}.
+   */
+  constructor(signature: string, callback: (args: Value[]) => string);
```

Comment:
I'm not very familiar with the JS API, so this is more of a placeholder. I would argue that this function should return a Sass AST node, but I'm not sure if such functionality is currently exposed through the JS API. If it is not, I would propose that it be impossible for users to construct mixin objects through the JS API, at least as part of an initial implementation.

Reply:
We don't want to make implementations beholden to exposing an entire Sass AST with a specific API. As far as the JS API and Embedded Protocol are concerned, a mixin should just be an opaque object that can't meaningfully be interacted with beyond returning it as-is. The JS constructor should be defined to just throw an error.